src/file/reader.rs

```diff
@@ -197,7 +232,7 @@ impl SerializedFileReader {
   }
 }
 
-impl FileReader for SerializedFileReader {
+impl<R: 'static + ParquetReader> FileReader<SerializedPageReader<R>, SerializedRowGroupReader<R>> for SerializedFileReader<R> {
```

Comment:
The `'static` lifetime is due to `ColumnReaderImpl` being generic over `DataType`

Reply:
But it is R that is static, isn't it?

Reply:
My point was whether or not we want to make `ParquetReader` bound to static. I am not sure about it.

Reply:
I think it should not be there. I've tried to get rid of the `'static` lifetime on `DataType`, but it's really hard.

Reply:
Yes it's hard to remove the `'static` constraint - I think the root cause is `get_decoder` returns boxed instance, which in default is supposed to have `static` lifetime, so that fields associated with `T` within the decoder can live beyond stack space. One possible way is to make the boxed instance have lifetime `'a` but then we'd have to populate `'a` over many places, which is not desirable. 

I think as long as we don't put any borrowed reference in `ParquetReader`, it should be OK.